### PR TITLE
docs: Add examples using network_state variable

### PR DESCRIPTION
**Pull request by liangwen12year**
Enhancement:

Reason:

Result:

Issue Tracker Tickets (Jira or BZ if any):

### CODE REVIEW

**Comment by liangwen12year** [comment]
@tyll , configure vrf interface is not supported in `network_connections`, besides that,  I also added other examples.

File: examples/network_state_example.yml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            - name: eth1
+              type: ethernet
+              state: up
+              ipv4:
+                enabled: true
+                dhcp: true
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+    - name: Configure the vrf interface test-vrf0 with vrf port eth1
```
Comment (liangwen12year) [inline examples/network_state_example.yml:207]: @eliaswimmer, this is the example of configuring vrf interface using the `network_state` variable.

**Comment by liangwen12year** [comment]
```
    - name: Configure the vrf interface test-vrf0 with vrf port eth1 and vrf route table 101
      vars:
        network_state:
          interfaces:
            - name: test-vrf0
              type: vrf
              state: up
              vrf:
                port:
                  - eth1
                route-table-id: 101
      ansible.builtin.include_role:
        name: linux-system-roles.network
```
@eliaswimmer, this is the example of configuring vrf interface using the network_state variable.

File: examples/network_state_example.yml [hunk 2]
```diff
@@ -0,0 +1,235 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Manager network using network_state
```
Comment (richm) [inline examples/network_state_example.yml:3]:
```suggestion
- name: Manage network using network_state
```

Comment (liangwen12year) [reply examples/network_state_example.yml]: thanks, I updated it.

**Review by richm — APPROVED** [review]

File: examples/network_state_example.yml [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    - name: Configure the vrf interface test-vrf0 with vrf port eth1 and vrf route table 101
+      vars:
+        network_state:
+          interfaces:
+            - name: test-vrf0
+              type: vrf
+              state: up
+              vrf:
+                port:
+                  - eth1
```
[the remark this answers is not in the document]
Comment (liangwen12year) [reply examples/network_state_example.yml:216]: yes, this is correct. 

```
[root@nmstate-test2 nmstate]# echo 'interfaces:
            - name: test-vrf0
              type: vrf
              state: up
              vrf:
                port:
                  - eth1
                route-table-id: 101' | nmstatectl apply - 
[2024-08-22T15:28:39Z INFO  nmstatectl] Nmstate version: 2.2.30
[2024-08-22T15:28:39Z INFO  nmstate::iface] Include interface eth1 to edit as its controller required so
[2024-08-22T15:28:39Z INFO  nmstate::query_apply::net_state] Created checkpoint /org/freedesktop/NetworkManager/Checkpoint/1
[2024-08-22T15:28:39Z INFO  nmstate::nm::query_apply::profile] Creating connection UUID Some("0c4e5ce3-8261-45d1-90a3-bd8800e9ee19"), ID Some("test-vrf0"), type Some("vrf") name Some("test-vrf0")
[2024-08-22T15:28:39Z INFO  nmstate::nm::query_apply::profile] Creating connection UUID Some("b1c6ca3c-9cdc-4e78-a72b-7d800cb8fa0a"), ID Some("eth1"), type Some("802-3-ethernet") name Some("eth1")
[2024-08-22T15:28:39Z INFO  nmstate::nm::query_apply::profile] Activating connection 0c4e5ce3-8261-45d1-90a3-bd8800e9ee19: test-vrf0/vrf
[2024-08-22T15:28:39Z INFO  nmstate::nm::query_apply::profile] Skip connection activation as its controller already activated its ports: b1c6ca3c-9cdc-4e78-a72b-7d800cb8fa0a: eth1/802-3-ethernet
[2024-08-22T15:28:39Z INFO  nmstate::query_apply::net_state] Destroyed checkpoint /org/freedesktop/NetworkManager/Checkpoint/1
[2024-08-22T15:28:39Z INFO  nmstate::iface] Include interface eth1 to edit as its controller required so
route-rules: {}
routes: {}
interfaces:
- name: test-vrf0
  type: vrf
  state: up
  vrf:
    port:
    - eth1
    route-table-id: 101
ovn: {}
```

**Pull request merged**